eclipse-che/che-theia
# build: Adapt Build System to upstream changes

Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>



### What does this PR do?
Provides alignment to upstream changes related to Build System.

Please pay attention that `linkless build` and classic `che-in-che development` were broken before.
The problems are NOT related to the current changes.
 I created separate issues 2 weeks ago:
- https://github.com/eclipse/che/issues/20814
- https://github.com/eclipse/che/issues/20816

After resolving those issues it will be possible to check if additional alignment is required for `linkless build` and classic `che-in-che` development.

For the classic `che-in-che` development probably some changes related to `jest` will be required (AFAIK there is a conflict with upstream - it looks like `jest` brings some transitive dependencies which leads to errors at compilation step).

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20538

### How to test this PR?

The PR doesn't provide any additional functionality.
It makes sense to check that it's possible to build `che-theia` assembly.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)



### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
